### PR TITLE
Fix generation of failures.csv when there are only non-outcome failures

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Fix a corner case from #144 that prevents the upload of `outcomes.csv` when there are failed jobs but no failed test cases (e.g. all tests that compile are passing, but there are configurations where the compilation fails).

Test run:
* development, release job (with Windows-2015 failing and causing the problem): [OpenCI](https://mbedtls.trustedfirmware.org/job/mbedtls-release-ci-testing/126/)
* 2.28, release job (all passing): [OpenCI](https://mbedtls.trustedfirmware.org/job/mbedtls-release-ci-testing/127/)

### CODE REVIEW

File: vars/analysis.groovy [hunk 1]
```diff
@@ -246,7 +246,7 @@ def process_outcomes() {
     // as test description or test suite was "FAIL".
     if (gen_jobs.failed_builds) {
         sh '''\
-LC_ALL=C grep ';FAIL;' outcomes.csv >"failures.csv"
+LC_ALL=C grep ';FAIL;' outcomes.csv >"failures.csv" || [ $? -eq 1 ]
```
Comment (mpg) [inline vars/analysis.groovy:249]: Ah, nice. I'd have written `|| true` but this is more precise, it only ignores the expected error where grep returns 1 for "no match". Learned something about robust `sh` scripting today :)

**Review by mpg — APPROVED** [review]
LGTM, thanks!

**Review by tom-cosgrove-arm — APPROVED** [review]

**Comment by mpg** [comment]
I checked the CI results and they look good: the development run has an empty `failures.csv` and a non-empty `outcomes.csv.xz` as expected.
